hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/ZKRMStateStore.java

```diff
@@ -938,28 +938,36 @@ private void handleApplicationAttemptStateOp(
         attemptStateDataPB.getProto().toByteArray();
     LOG.debug("{} info for attempt: {} at: {}", operation, appAttemptId, path);
 
-    switch (operation) {
-    case UPDATE:
-      if (exists(path)) {
-        zkManager.safeSetData(path, attemptStateData, -1, zkAcl,
-            fencingNodePath);
-      } else {
-        zkManager.safeCreate(path, attemptStateData, zkAcl,
-            CreateMode.PERSISTENT, zkAcl, fencingNodePath);
-        LOG.debug("Path {} for {} didn't exist. Created a new znode to update"
-            + " the application attempt state.", path, appAttemptId);
+    try {
+      switch (operation) {
+      case UPDATE:
+        if (exists(path)) {
+          zkManager.safeSetData(path, attemptStateData, -1, zkAcl,
+              fencingNodePath);
+        } else {
+          zkManager.safeCreate(path, attemptStateData, zkAcl,
+              CreateMode.PERSISTENT, zkAcl, fencingNodePath);
+          LOG.debug("Path {} for {} didn't exist. Created a new znode to update"
+              + " the application attempt state.", path, appAttemptId);
 
+        }
+        break;
+      case STORE:
+        zkManager.safeCreate(path, attemptStateData, zkAcl, CreateMode.PERSISTENT,
+            zkAcl, fencingNodePath);
+        break;
+      case REMOVE:
+        zkManager.safeDelete(path, zkAcl, fencingNodePath);
+        break;
+      default:
+        break;
+      }
+    } catch (KeeperException.NoNodeException nne){
+      if(!exists(path)){
```

Comment:
Here, You just need to focus on the NoNodeException of REMOVE action

Reply:
Thank you. I've modified and tested this.